lib/stable_email.py

```diff
@@ -0,0 +1,178 @@
+        msg = is_greg_request(commit)
+        if msg:
+            print("Found: %s" % commit.summary)
+            fg[msg["message-id"]] = {"request": commit}
+        x += 1
+
+    return fg
+
+
+def get_review_replies(oldest, fg):
+    print("* Looking for replies...")
+    x = 0
+    repo = git.Repo("/data/lore-stable")
+    commits = repo.iter_commits("HEAD")
+
+    for commit in commits:
+        # Limit search
+        if is_beyond_time_search(commit, oldest):
+            print("Done. (Looked at %d messages.)" % x)
+            break
+
+        msg = commit_to_email_message(commit)
+
+        if "in-reply-to" in msg and msg["in-reply-to"] in fg:
+            inrt = msg["in-reply-to"]
+            efrom = msg["from"]
+            # if 'gregkh' not in efrom:
+            if "linaro.org" in efrom:
+                print("%d: %s" % (x, efrom))
+                if "replies" in fg[inrt]:
+                    fg[inrt]["replies"].append(commit)
+                else:
+                    fg[inrt]["replies"] = [commit]
+
+        x += 1
+
+    # print(str(from_greg))
+    return fg
+
+
+class Review(object):
+    request = None
+    reply = None
+    elapsed_time = None
+
+    def __init__(self, req, reply):
+        self.request = req
+        self.reply = reply
+
+    def calc_elapsed_time(self):
+        request_time = self.request.committed_datetime
+        if self.reply:
+            reply_time = self.reply.committed_datetime
+
+            # This is a datetime.timedelta
+            self.elapsed_time = reply_time - request_time
+
+    def get_elapsed_time(self):
+        self.calc_elapsed_time()
+
+        et = self.elapsed_time
+        if et:
+            days = self.elapsed_time.days
+            hours = days * 24
+            hours += int(self.elapsed_time.seconds / 3600)
+            minutes = int((self.elapsed_time.seconds % 3600) / 60)
+            return "%d:%02d" % (hours, minutes)
+        else:
+            return None
+
+    def get_sla_mark(self):
+        self.calc_elapsed_time()
+        et = self.elapsed_time
+        if et:
+            hours = et.seconds / 3600
+            if et.days >= 2:
+                return ">48h"
+            elif et.days < 2 and et.days >= 1:
+                return "<48h"
+            elif et.days < 1 and hours >= 8:
+                return "<24h"
+            else:
+                return "<8h"
+        else:
+            return None
+
+    def get_linux_version(self):
+        sub = self.request.summary
+        if not sub.endswith("-stable review"):
+            return None
+        if not sub.startswith("["):
+            return None
+        sub1 = sub.replace("-stable review", "")
+        return sub1[sub1.rfind(" ") + 1 :]
+
+    def get_ymd(self):
+        return self.request.committed_datetime.strftime("%Y-%m-%d")
+
+    def get_from(self):
+        return "%s <%s>" % (self.reply.author.name, self.reply.author.email)
+
+    def get_id(self):
```

Comment:
this could be generalized and perhaps more readable if implemented as:

```
def get_header(header):
    return commit_to_email_message(self.reply).get(header, None)
get_header('mssage-id')
```
Then if you do still want a message-id specific call, it could call get_header for message-id - but i think calling the function "get_id" is ambiguous.